### PR TITLE
Implement updateApplication API call

### PR DESCRIPTION
**Pull request by taran-gill**
[See the PR for updateApplication](https://github.com/deltahacks/deltahacks-secrets/pull/15).

We should restrict people's ability to edit their apps _before_ we merge this in so that somebody can't accidentally upload their app (removing the Login button on the landing page might be sufficient tbh)

### CODE REVIEW

File: my/src/views/Apply.vue [hunk 1]
```diff
@@ -206,42 +207,36 @@ export default Vue.extend({
...
+        } else {
+          this.snack.message = submitting ? 'Application submitted' : 'Progress saved!';
         this.snack.color = 'success';
-        submit = true;
-      } else if (
-        submitting &&
-        this.app._.status === 'in progress' &&
-        !verified
-      ) {
-        this.snack.message = 'Please verify your email before submitting!';
```
Comment (taran-gill) [inline my/src/views/Apply.vue:231]: We're verifying emails on the back-end instead 🎉

Comment (taran-gill) [reply my/src/views/Apply.vue:231]: Oh, and the '_' property is basically read-only for the front-end, so there's a bunch of dead code we'll have to get rid of. I'll do that later to keep this PR small

**Review by ShubhSehgal — APPROVED** [review]
Looks good!

**Review by amirafzali — APPROVED** [review]
Everything LGTM!

**Pull request merged**